### PR TITLE
Allow app key management of other users, including key scopes

### PR DESCRIPTION
**Pull request by jorhett**
putting this just in our repo for now, will create PR in their repo later

### CODE REVIEW

File: datadog/resource_datadog_application_key.go [hunk 1]
```diff
@@ -104,7 +154,7 @@ func resourceDatadogApplicationKeyUpdate(ctx context.Context, d *schema.Resource
 	apiInstances := providerConf.DatadogApiInstances
 	auth := providerConf.Auth
 
-	resp, httpResponse, err := apiInstances.GetKeyManagementApiV2().UpdateCurrentUserApplicationKey(auth, d.Id(), *buildDatadogApplicationKeyUpdateV2Struct(d))
+	resp, httpResponse, err := apiInstances.GetKeyManagementApiV2().UpdateApplicationKey(auth, d.Id(), *buildDatadogApplicationKeyUpdateV2Struct(d))
```
Comment (jorhett) [inline datadog/resource_datadog_application_key.go:157]: @chicks-net if we're going to keep the existing resources working, we need to add code just before this that modifies `d` to include the owner (user_id) of the application key being used -- only if owner is empty.

```
	if d.Get("owner") == "" {
	  d.Set("owner") = auth.I.assume.we.can.get.the.user.from.this?
	}
```

Comment (chicks-net) [reply datadog/resource_datadog_application_key.go:157]: I don't think this is a golang issue.  I pushed a fix in https://github.com/adRise/adrise_infrastructure/pull/9649/commits/de82505379c5f8a55a78f5f77906997bf45bfeaf that cleans up the terraform plan for `datadog_application_key.eng_infra_terraform`.

Comment (jorhett) [reply datadog/resource_datadog_application_key.go:157]: I know, I did that too. But the problem is that instantly every existing TF resource out there will see a state change, so this breaks existing users. If we look up the existing user id and add it to the resource then there will be no change.

Comment (chicks-net) [reply datadog/resource_datadog_application_key.go:157]: 1. Do we have any existing examples of that?  I'm not seeing `datadog_application_key` used anywhere else.
2. Isn't that lookup what is already being done in the lower half of `updateApplicationKeyState()`?

**Pull request closed without merging**